utils/catalog_production/add_glsn_properties.py

```diff
@@ -0,0 +1,286 @@
+            sed_wave_width.append(int(label_split[2])/10)
+    bin_order = np.argsort(sed_min_wave)
+    sed_label = np.array(sed_label)[bin_order]
+    sed_min_wave = np.array(sed_min_wave)[bin_order]
+    sed_wave_width = np.array(sed_wave_width)[bin_order]
+
+    for gal_id, gal_z in zip(gcr_glsn_match[:, 1], 
+                             gcr_glsn_match[:, 0]):
+        
+        if i % 100 == 0:
+            print(i)
+        i+=1
+        #print(np.int(gal_id), type(data_df['galaxy_id'].iloc[0]))
+        #print(gal_z, type(gal_z), data_df['redshift_true'].iloc[0])
+        #print(int(gal_id))
+        data_subset = data_df.query(str('galaxy_id == %i' % np.int(gal_id)))# + ' and ' + 'redshift_true == {}'.format(gal_z))) ## Galaxy Ids are not unique in cosmoDC2_v0.1
+        #print(data_subset)
+        mag_array = []
+        lsst_mag_array = [data_subset['mag_%s_lsst' % band_name].values[0] for band_name in ['u', 'g', 'r', 'i', 'z', 'y']]
+        for sed_bin in sed_label:
+            mag_array.append(-2.5*np.log10(data_subset[sed_bin].values[0]))
+        mag_array = np.array(mag_array)
+        lsst_mag_array = np.array(lsst_mag_array)
+        lsst_mags.append(lsst_mag_array)
+        mag_30_list.append(mag_array)
+        redshift_list.append(gal_z)
+    
+    mag_30_list = np.array(mag_30_list).T
+    lsst_mags = np.array(lsst_mags).T
+    redshift_list = np.array(redshift_list)
+
+    print('Getting SEDs')
+
+    sed_name, magNorm, av, rv = sed_from_galacticus_mags(mag_30_list, redshift_list,
+                                                         H0, Om0, sed_min_wave, sed_wave_width, lsst_mags)
+
+    return sed_name, magNorm, av, rv
+
+
+if __name__ == "__main__":
+
+    catalog_version = sys.argv[1]
+    glsn_cat = sys.argv[2]
+
+    dc2_df_system = pd.read_hdf(glsn_cat, key='system')
+    dc2_df_image = pd.read_hdf(glsn_cat, key='image')
+
+    catalog = GCRCatalogs.load_catalog('%s_small' % catalog_version)
+
+    load_limited_cat(dc2_df_system, catalog)
+    data_df = pd.read_csv('sn_matching_checkpoint_1.csv')
+    print('Saved Checkpoint 1.')
+    match_up_glsn(dc2_df_system, data_df)
+    get_catalog_mags(catalog)
+    print('Saved Checkpoint 2.')
+
+    data_df = pd.read_csv('sn_matching_checkpoint_2.csv')
+    gcr_glsn_match = np.genfromtxt('gcr_glsn_match.txt')
+
+    sed_name, magNorm, av, rv = get_30_band_mags(gcr_glsn_match, data_df, catalog)
+
+    sed_name_array = np.array(sed_name)
+    magNorm_array = np.array(magNorm)
+    av_array = np.array(av)
+    rv_array = np.array(rv)
+
+    np.savetxt('sed_name_list_sn.txt', sed_name_array, fmt='%s')
+    np.savetxt('magNorm_list_sn.txt', magNorm_array)
+    np.savetxt('av_list_sn.txt', av_array)
+    np.savetxt('rv_list_sn.txt', rv_array)
+
+    test_bandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+    imsimband = Bandpass()
+    imsimband.imsimBandpass()
+
+    mag_norm_glsne = []
+    for i, idx in list(enumerate(keep_rows)):
```

Comment:
I don't think `keep_rows` is defined in this scope.  It is internal to `match_up_glsn()`.